zarr/codecs.py

```diff
@@ -485,6 +485,105 @@ def __repr__(self):
 codec_registry[Delta.codec_id] = Delta
 
 
+class AsType(Codec):
+    """Filter to convert data between different types.
+
+    Parameters
+    ----------
+    dtype : dtype
+        Data type used to store the data.
+    astype : dtype, optional
+        Data type of the data read in.
+
+    Notes
+    -----
+    If `astype` is an integer data type, please ensure that it is
+    sufficiently large to store encoded values. No checks are made and data
+    may become corrupted due to integer overflow if `astype` is too small.
+    Note also that the encoded data for each chunk includes the absolute
+    value of the first element in the chunk, and so the encoded data type in
+    general needs to be large enough to store absolute values from the array.
+
+    Examples
+    --------
+    >>> import zarr
+    >>> import numpy as np
+    >>> x = np.arange(100, 120, 2, dtype='i1')
+    >>> x
+    array([100, 102, 104, 106, 108, 110, 112, 114, 116, 118], dtype=int8)
+    >>> f = zarr.AsType(dtype=x.dtype, astype='i8')
+    >>> y = f.decode(x)
+    >>> y
+    array([100, 102, 104, 106, 108, 110, 112, 114, 116, 118])
+    >>> z = f.encode(y)
+    >>> z
+    array([100, 102, 104, 106, 108, 110, 112, 114, 116, 118], dtype=int8)
+
+    """  # flake8: noqa
+
+    codec_id = 'astype'
+
+    def __init__(self, dtype, astype=None):
```

Comment:
What about something like:

```
Notes
-----
This method returns a new Array object which is a view on the same underlying 
chunk data. Modifying any data via the view will also modify data in the original 
array. This is an experimental feature and care may be needed to avoid data 
corruption or unexpected loss of precision.

See Also
--------
Array.view
```